sw/airborne/firmwares/rotorcraft/autopilot_firmware.c

```diff
@@ -234,9 +248,7 @@ void autopilot_event(void)
       || autopilot.mode == AP_MODE_FAILSAFE
 #endif
      ) {
-    struct NedCoor_f *accel = stateGetAccelNed_f();
-    if (accel->z < -THRESHOLD_GROUND_DETECT ||
-        accel->z > THRESHOLD_GROUND_DETECT) {
+    if (autopilot_ground_detection()) {
```

Comment:
Why does the ground detection in ```NAV``` need to be different from the ground detection to stop ```in_flight```? PS: Detecting ground only using ```accel->z``` is super risky.

Reply:
Here, the idea was to detect the shock when hitting the ground from spike on the vertical acceleration. It allows a "faster" detection of the ground than waiting for multiple factors to be valid.
It is super risky indeed, but not activated by default. This is something you should activate only for the final flare, when already close to the ground, and if you have a really use for it.
In any case, it should either stay the same than before, or completely removed if not desired anymore.